tests/unit/test_analytics.py

```diff
@@ -88,7 +86,72 @@ def test_find_periods(
         (dummy_datetime_list[9], dummy_datetime_list[9]),
     ]
 
-    result = sleep_detection._find_periods(window_measurement)
+    result = analytics._find_periods(window_measurement)
+
+    assert result == expected_result, f"Expected {expected_result}, but got {result}"
+
+
+@pytest.mark.parametrize(
+    "non_wear_array, expected_result",
+    [
+        (np.array([0, 1, 1, 1, 0]), []),
+        (np.array([0, 0, 0, 1, 1]), []),
+        (np.array([0, 0, 0, 1, 0]), []),
+    ],
+)
+def test_remove_nonwear_periods_overlap(
+    non_wear_array: np.ndarray, expected_result: List
+) -> None:
+    """Test the _remove_nonwear_from_sleep method.
+
+    This test is for the following cases:
+        - where the non-wear period overlaps with sleep onset.
+        - where the non-wear period overlaps with sleep wakeup.
+        - where the non-wear period is within the sleep window.
+    """
+    dummy_date = datetime(2024, 5, 2)
```

Comment:
where did this datetime come from? :)

Reply:
fixed (we might need to do a thorough review through all unit tests at some point)